app/controllers/idv/cancellations_controller.rb

```diff
@@ -33,6 +35,27 @@ def destroy
 
     private
 
+    def barcode_step
+      return params[:step] == 'barcode'
+    end
+
+    def enrollment
+      return InPersonEnrollment.where(user_id: current_user.id).first
+    end
+
+    def extra_analytics_attributes
+      {
+        cancelled_enrollment: false,
+        enrollment_code: enrollment.enrollment_code,
+        enrollment_id: enrollment.id,
```

Comment:
We might want to be careful about expecting `enrollment` to exist, since we don't validate that anywhere. For example, I was able to trigger a 500 error by;

1. Sign in
2. Go to http://localhost:3000/verify/cancel?step=barcode
3. Click "No, keep going"

Reply:
good catch!